cargo-dylint/tests/ci.rs

```diff
@@ -133,6 +133,169 @@ fn cargo_dylint_and_dylint_readmes_are_equal() {
+            .map(|(name, _)| format!("[`{name}`](./{category}/{name})").len())
+            .max()
+            .unwrap_or(0);
+
+        let max_desc_width = examples
+            .iter()
+            .map(|(_, description)| description.len())
+            .max()
+            .unwrap_or(0);
+
+        let example_col_width = max(example_header.len(), max_example_width);
+        let desc_col_width = max(desc_header.len(), max_desc_width);
+
+        // Write header
+        write!(content, "\n## {}\n\n", capitalize(category)).unwrap();
+        #[allow(clippy::uninlined_format_args)]
+        writeln!(
+            content,
+            "| {:<example_col_width$} | {:<desc_col_width$} |",
+            example_header, desc_header
+        )
+        .unwrap();
+        writeln!(
+            content,
+            "| {:-<example_col_width$} | {:-<desc_col_width$} |",
+            "", ""
+        )
+        .unwrap(); // Separator line
+
+        // Write rows with padding
+        for (name, description) in examples {
+            let example_link = format!("[`{name}`](./{category}/{name})");
+            #[allow(clippy::uninlined_format_args)]
+            writeln!(
+                content,
+                "| {:<example_col_width$} | {:<desc_col_width$} |",
+                example_link, description
+            )
+            .unwrap();
+        }
+    }
+
+    content
+}
+
+fn extract_between_markers(content: &str) -> Option<String> {
+    const START_MARKER: &str = "<!-- lint descriptions start -->";
+    const END_MARKER: &str = "<!-- lint descriptions end -->";
+    let start = content.find(START_MARKER)? + START_MARKER.len();
+    let end = content.find(END_MARKER)?;
+    Some(content[start..end].trim().to_string())
+}
+
+fn collect_examples_from_category(examples_dir: &Path, category: &str) -> Vec<(String, String)> {
+    let mut examples = Vec::new();
+    let category_dir = examples_dir.join(category);
+
+    for entry in read_dir(&category_dir).unwrap() {
+        let entry = entry.unwrap();
+        let metadata = entry.metadata().unwrap();
+        if metadata.is_dir() {
+            let cargo_toml_path = entry.path().join("Cargo.toml");
+            if cargo_toml_path.exists() {
+                if let Some((name, desc)) = extract_name_and_description(&cargo_toml_path) {
+                    examples.push((name, desc));
+                }
+            }
+        }
+    }
+
+    // Sort examples by name
+    examples.sort_by(|(a, _), (b, _)| a.cmp(b));
+    examples
+}
+
+fn extract_name_and_description(cargo_toml_path: &Path) -> Option<(String, String)> {
+    let content = read_to_string(cargo_toml_path).ok()?;
+
+    // Get the name from the directory
+    let name = cargo_toml_path
+        .parent()
+        .and_then(|path| path.file_name())
+        .unwrap()
+        .to_string_lossy()
+        .to_string();
+
+    // Extract the description using regex
+    let re = Regex::new(r#"description\s*=\s*"([^"]*)""#).unwrap();
```

Comment:
Since `extract_name_and_description` will be called multiple times, could you please make this a static using `LazyLock`?

Here are a couple of examples: https://github.com/trailofbits/dylint/blob/d2876b4fb978a13daf0c0c7c24bc307207fcb82e/examples/supplementary/commented_code/src/lib.rs#L102-L105